sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
> (if so I'd figure `UTF-8`)

I thought the same thing but `UTF-8` was giving errors when running the test and after trying other encodings `latin-1` is the one that got the tests to pass.

> If the latter, the interface should just be bytes and the application must handle the encoding/decoding. 

> Also also, we should be using the compatibility libraries that provide str and bytes as unique types in py2.

I think switching to a compatibility library to make the py2 version have a separate bytes object that it manipulates could cause some serious backwards compatibility issues.

> Also, encoding and decoding are still relevant in py2 since the meaning of a given set of bytes still varies depending on the coding used. https://repl.it/JRTE

This is moving between strings and unicode strings instead of bytes. I think the reason to leave it be here is the interface that it's using takes the `str` object with the bytes already.